runtime/include/tasks/qthreads/tasks-qthreads.h

```diff
@@ -118,33 +118,24 @@ volatile int chpl_qthread_done_initializing;
 
 typedef struct {
   chpl_taskID_t id;
+  // lock_filename, lock_lineno
 } chpl_task_bundleData_t;
 
-typedef struct {
-    int32_t task_filename;
-    int task_lineno;
-    chpl_taskID_t id;
-    chpl_bool is_executeOn;
-    c_sublocid_t requestedSubloc;  // requested sublocal for task
-    chpl_task_prvData_t prvdata;
-} chpl_task_prvDataImpl_t;
-
-// Define PRV_DATA_IMPL_VAL to set up a chpl_task_prvData_t.
-#define PRV_DATA_IMPL_VAL(_fn, _ln, _id, _is_execOn, _subloc, _serial) \
-        { .task_filename = _fn, \
-          .task_lineno = _ln, \
-          .id = _id, \
-          .is_executeOn = _is_execOn, \
-          .requestedSubloc = _subloc, \
-          .prvdata = { .serial_state = _serial } }
+// Now that chpl_task_bundleData_t is defined, include
+// chpl-tasks-bundle.h so the full structure is available
+// in the rest of this file.
+#include "chpl-tasks-bundle.h"
 
 // Structure of task-local storage
 typedef struct chpl_qthread_tls_s {
-    /* Task private data: serial state, etc. */
-    chpl_task_prvDataImpl_t chpl_data;
-    /* Reports */
-    int32_t    lock_filename;
-    size_t      lock_lineno;
+  chpl_task_bundle_p bundle;
+  // The below fields could move to chpl_task_bundleData_t
+  // That would reduce the size of the task local storage,
+  // but increase the size of executeOn bundles.
```

Comment:
Since qthreads doesn't (apparently) give a way to get the arguments provided when starting the task, we still have to stash a pointer to the task bundle. I'd expect it is clearest code if chpl_qthread_tls_t continues to have that pointer in it. Then, it's just a matter of whether or not that type has prvdata, lock_filename, and lock_lineno; or if those are stored in the argument bundle (even though they are never used to pass data).